### PR TITLE
[FLINK-19734][table-planner-blink] Replace 'collection' connector by 'values' connector for temporal join plan tests

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

The 'collection' connector is a legacy connector, we should use the 'values' connector which is based on FLIP-95 interfaces. 

## Brief change log

- Replace 'collection' connector by 'values' connector in plan tests
- Decouple scan source and lookup source for values connector

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
cc @leonardBang

File: flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/planner/plan/stream/sql/join/TemporalJoinTest.xml [hunk 1]
```diff
@@ -43,13 +43,13 @@ Calc(select=[amount, currency, rowtime, PROCTIME_MATERIALIZE(proctime) AS procti
...
    :     +- Calc(select=[amount, currency, rowtime, PROCTIME() AS proctime])
-   :        +- LegacyTableSourceScan(table=[[default_catalog, default_database, Orders, source: [CollectionTableSource(amount, currency, rowtime)]]], fields=[amount, currency, rowtime])
+   :        +- TableSourceScan(table=[[default_catalog, default_database, Orders]], fields=[amount, currency, rowtime])
    +- Exchange(distribution=[hash[currency]])
       +- Calc(select=[currency, rate, rowtime], where=[<(rate, 100)])
          +- Rank(strategy=[AppendFastStrategy], rankType=[ROW_NUMBER], rankRange=[rankStart=1, rankEnd=1], partitionBy=[currency], orderBy=[rowtime DESC], select=[currency, rate, rowtime])
             +- Exchange(distribution=[hash[currency]])
                +- WatermarkAssigner(rowtime=[rowtime], watermark=[rowtime])
-                  +- LegacyTableSourceScan(table=[[default_catalog, default_database, RatesHistory, source: [CollectionTableSource(currency, rate, rowtime)]]], fields=[currency, rate, rowtime])
+                  +- TableSourceScan(table=[[default_catalog, default_database, RatesHistory]], fields=[currency, rate, rowtime])
```
Comment (leonardBang) [inline flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/planner/plan/stream/sql/join/TemporalJoinTest.xml:52]: Could we add one `Collection` test to ensure the temporal join plan works fine with Legacy connector?

**Review by leonardBang — COMMENTED** [review]
Thanks @wuchong for the contribution, the PR generally looks fine,  I left one minor comment.

**Comment by wuchong** [comment]
Thanks for the reviewing @leonardBang . I have added tests for legacy source.

**Review by leonardBang — APPROVED** [review]
@wuchong Thanks for the update, LGTM

**Comment by wuchong** [comment]
Thanks for the reviewing. 

Merging...

**Pull request merged**